cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTableAdmin.java

```diff
@@ -141,13 +174,56 @@ private boolean tableExistsInternal(Connection connection, StructuredTableId tab
     }
   }
 
+  // Add new non-primary key columns to table
+  private void addColumns(Connection connection, StructuredTableSchema schema)
+    throws IOException, TableNotFoundException {
+    StructuredTableId tableId = schema.getTableId();
+    try {
+      Optional<String> columnStatement = getAddColumnStatement(tableId,
+                                                               connection.getMetaData(), schema);
+      if (!columnStatement.isPresent()) {
```

Comment:
Actually will this ever happen since you already checked for new and old schemas are not equals? I suspected this should actually be an exception?

Reply:
It might be adding index only?